engines/sci/graphics/view.cpp

```diff
@@ -681,39 +655,40 @@ void GfxView::unpackCel(int16 loopNo, int16 celNo, byte *outPtr, uint32 pixelCou
 	}
 }
 
-const byte *GfxView::getBitmap(int16 loopNo, int16 celNo) {
-	loopNo = CLIP<int16>(loopNo, 0, _loopCount -1);
-	celNo = CLIP<int16>(celNo, 0, _loop[loopNo].celCount - 1);
+const SciSpan<const byte> &GfxView::getBitmap(int16 loopNo, int16 celNo) {
+	loopNo = CLIP<int16>(loopNo, 0, _loop.size() - 1);
+	celNo = CLIP<int16>(celNo, 0, _loop[loopNo].cel.size() - 1);
 	if (_loop[loopNo].cel[celNo].rawBitmap)
-		return _loop[loopNo].cel[celNo].rawBitmap;
+		return *_loop[loopNo].cel[celNo].rawBitmap;
 
 	uint16 width = _loop[loopNo].cel[celNo].width;
 	uint16 height = _loop[loopNo].cel[celNo].height;
 	// allocating memory to store cel's bitmap
-	int pixelCount = width * height;
-	_loop[loopNo].cel[celNo].rawBitmap = new byte[pixelCount];
-	byte *pBitmap = _loop[loopNo].cel[celNo].rawBitmap;
+	const uint pixelCount = width * height;
+	_loop[loopNo].cel[celNo].rawBitmap.allocate(pixelCount, _resource->getDescription() + Common::String::format(" loop %d cel %d", loopNo, celNo));
+	SciSpan<byte> rawBitmap = *_loop[loopNo].cel[celNo].rawBitmap;
```

Comment:
This construction is a bit awkward. It might be cleaner to allocate the memory manually with `malloc` here, and then add a constructor to create a `SpanOwner` (and `Span`) from that to avoid the implicit const-casting.